Application/EdFi.Ods.AdminApi/Infrastructure/WebApplicationBuilderExtensions.cs

```diff
@@ -1,237 +1,239 @@
-            DbConfiguration.SetConfiguration(new DatabaseEngineDbConfiguration(Common.Configuration.DatabaseEngine.Postgres));
-
-            webApplicationBuilder.Services.AddDbContext<AdminApiDbContext>(
-                options =>
-                {
-                    options.UseNpgsql(adminConnectionString);
-                    options.UseOpenIddict<ApiApplication, ApiAuthorization, ApiScope, ApiToken, int>();
-                });
-
-            webApplicationBuilder.Services.AddScoped<EdFi.SecurityCompatiblity53.DataAccess.Contexts.ISecurityContext>(
-                sp => new EdFi.SecurityCompatiblity53.DataAccess.Contexts.PostgresSecurityContext(securityConnectionString));
-
-            webApplicationBuilder.Services.AddScoped<ISecurityContext>(
-                sp => new PostgresSecurityContext(securityConnectionString));
-
-            webApplicationBuilder.Services.AddScoped<IUsersContext>(
-                sp => new PostgresUsersContext(adminConnectionString));
-
-            return (adminConnectionString, false);
-        }
-
-        if (DatabaseEngineEnum.Parse(databaseEngine).Equals(DatabaseEngineEnum.SqlServer))
-        {
-            DbConfiguration.SetConfiguration(new DatabaseEngineDbConfiguration(Common.Configuration.DatabaseEngine.SqlServer));
-
-            webApplicationBuilder.Services.AddDbContext<AdminApiDbContext>(
-                options =>
-                {
-                    options.UseSqlServer(adminConnectionString);
-                    options.UseOpenIddict<ApiApplication, ApiAuthorization, ApiScope, ApiToken, int>();
-                });
-
-            webApplicationBuilder.Services.AddScoped<EdFi.SecurityCompatiblity53.DataAccess.Contexts.ISecurityContext>(
-                sp => new EdFi.SecurityCompatiblity53.DataAccess.Contexts.SqlServerSecurityContext(securityConnectionString));
-
-            webApplicationBuilder.Services.AddScoped<ISecurityContext>(
-                sp => new SqlServerSecurityContext(securityConnectionString));
-
-            webApplicationBuilder.Services.AddScoped<IUsersContext>(
-                sp => new SqlServerUsersContext(adminConnectionString));
-
-            return (adminConnectionString, true);
-        }
-
-        throw new Exception($"Unexpected DB setup error. Engine '{databaseEngine}' was parsed as valid but is not configured for startup.");
-    }
-
-    private enum HttpVerbOrder
-    {
-        GET = 1,
-        POST = 2,
-        PUT = 3,
-        DELETE = 4,
-    }
-}
+        webApplicationBuilder.Services.AddSingleton<IOdsSecurityModelVersionResolver>(sp =>
+        {
+            var apiServerUrl = webApplicationBuilder.Configuration.GetValue<string>("AppSettings:ProductionApiUrl");
+            var using53Cqe = webApplicationBuilder.Configuration.GetValue<bool>("AppSettings:Using53Cqe");
+
+            var validator = sp.GetRequiredService<IOdsApiValidator>();
+            return new OdsSecurityVersionResolver(validator, apiServerUrl, using53Cqe);

```

Comment:
This contains the only non-line ending change in the file: injection of the new appSettings value.